### ISSUE TITLE
Application is killed when backgrounded and immersive mode enabled.

### ISSUE THREAD

**Issue by codegist** [libgdx/libgdx#3861]
I've noticed my application is getting constantly killed when I background it since when I started using immersiveMode.
I started setting immersive mode manually before noticing the useImmersiveMode flag in the android config. Both of them end up with the application getting killed so I suppose the problem is not in Ligbdx immersive mode implementation but rather on how it deals with it. The application gets killed by the dead-lock workaround in AndroidGraphics.pause() method.

I tried running a AndroidLauncher with a blank ApplicationAdapter in it (so no game whatsoever). I slide on the right to get the virtual button bar to appear and tap Home. Application is backgrounded, and after a few seconds (4 as per AndroidGraphics.pause()), the application is killed.

Also with a game with immersive and sounds/music on, backgrounding it will continue playing the sounds (until it gets killed)

"waiting for pause synchronization took too long; assuming deadlock and killing"

**Comment by codegist** [comment]
Just for reference for whoever gets that problems, I'm using myself a very quick and very very dirty patch until it gets fixed. Here it is:

```
package com.badlogic.gdx.backends.android;

import java.util.concurrent.ExecutorService;
import java.util.concurrent.Executors;

public class PatchedAndroidApplication extends AndroidApplication {

	private final ExecutorService exec = Executors.newSingleThreadExecutor();
	private final Runnable forcePause = new Runnable() {
		@Override
		public void run() {
			try {Thread.sleep(100);} catch (InterruptedException e) {}
			graphics.onDrawFrame(null);
		}
	};

	@Override
	protected void onPause () {
		if(useImmersiveMode) {
			exec.submit(forcePause);
		}
		super.onPause();
	}
}
```

Just change your `AndroidLauncher` to extends `PatchedAndroidApplication` rather than `AndroidApplication`.

**Comment by Metaphore** [comment]
I confirm, same issue appears in my project too. I've tried turn off immersive mode and it works fine. @codegist thanks a lot for that dirty workaround!

**Comment by pravdokop** [comment]
The same situation for me: In the immersive mode if I press Home button the application is killed with the message "waiting for pause synchronization took too long; assuming deadlock and killing". Will some one from libGDX team work on solving this bug???

**Comment by tanaya177** [comment]
This work around increases ANR.

**Comment by cypherdare** [comment]
This theoretically should be fixed by #6074. I'm going to test if it's fixed in 1.9.11 on one of my projects that was exhibiting the issue for the next couple weeks and close this if it looks good.

**Comment by cypherdare** [comment]
Testing with 1.9.11 was successful.
